packages/smart-contracts/src/lib/ContractArtifact.ts

```diff
@@ -94,6 +94,24 @@ export class ContractArtifact<TContract extends Contract> {
     }));
   }
 
+  /**
+   * Retrieve all addresses for all versions for all networks
+   * @returns the addresses of the deployed contract and the associated network and version.
+   */
+  getAllAddressesFromAllNetworks(): { version: string; address: string; networkName: string }[] {
```

Comment:
No enum or string union for the `networkName`?

Reply:
not sure that we have this ☹️

Reply:
fixed in https://github.com/RequestNetwork/requestNetwork/pull/1056